tests/phpunit/tests/fonts/font-library/wpFontsDir.php

```diff
@@ -69,4 +81,57 @@ function set_new_values( $defaults ) {
 
 		$this->assertSame( static::$dir_defaults, $font_dir, 'The wp_get_font_dir() method should return the default values.' );
 	}
+
+	public function test_should_created_fonts_dir_in_uploads_when_fails_in_wp_content() {
+		// Set the expected results.
+		$upload_dir = wp_upload_dir();
+		$path       = path_join( $upload_dir['basedir'], 'fonts' );
+		$expected   = array(
+			'path'    => $path,
+			'url'     => $upload_dir['baseurl'] . '/fonts',
+			'subdir'  => '',
+			'basedir' => $path,
+			'baseurl' => $upload_dir['baseurl'] . '/fonts',
+			'error'   => false,
+		);
+
+		$this->fake_no_new_directories_in_wp_content();
+		$this->assertFileExists( self::$fake_fonts_file );
+
+		$font_dir = wp_get_font_dir();
+
+		$this->assertDirectoryDoesNotExist( path_join( WP_CONTENT_DIR, 'fonts' ), 'The `wp-content/fonts` directory should not exist' );
+		$this->assertDirectoryExists( $font_dir['path'], 'The `uploads/fonts` directory should exist' );
+		$this->assertSame( $expected, $font_dir, 'The font directory should be a subdir in the uploads directory.' );
+	}
+
+	private function remove_fonts_directory() {
+		$directories = array(
+			path_join( WP_CONTENT_DIR, 'fonts' ),
+			path_join( WP_CONTENT_DIR, 'uploads/fonts' ),
+			'/custom-path/fonts/my-custom-subdir',
+		);
+
+		foreach ( $directories as $dir ) {
+			if ( ! is_dir( $dir ) ) {
+				continue;
+			}
+
+			$this->rmdir( $dir );
+			@rmdir( $dir );
+		}
+	}
+
+	private function fake_no_new_directories_in_wp_content() {
```

Comment:
This is a weird fake, but it works. The way it works is this:

A file named `fonts` that exists in the `wp-content` will trigger `wp_mkdir_p()` to fail into the `file_exists()` bail out and return false when `is_dir()` runs.

That will then cause the fallback branch to execute.

Reply:
I created it in a method for 2 reasons:

1. In the test, makes it more readable and less code.
2. Reusability. This logic might be useful elsewhere for other Font Library tests. If yes, it can be moved to a trait or base TestCase.

Reply:
Wondering if the method name should be focused on what it does -> makes `wp_mkdir_p()` return `false` ??

Reply:
I think the name is fine

Reply:
I renamed this function to `create_fake_file_to_avoid_dir_creation` because I think it's a little bit cleared, and making it path-agnostic was needed because it is used to create a path received as a parameter in the additional test cases added here: https://github.com/WordPress/wordpress-develop/pull/6259/commits/90a3982636f1e47209e19a8a170ce027fb921af0